src/client/address_book/impl.cc

```diff
@@ -444,24 +444,25 @@ std::unique_ptr<const kovri::core::IdentHash> AddressBook::GetLoadedAddressIdent
 // Used only by HTTP Proxy
 void AddressBook::InsertAddressIntoStorage(
     const std::string& address,
-    const std::string& base64) {
-  kovri::core::IdentityEx ident;
+    const std::string& base64)
+{
   try
     {
+      kovri::core::IdentityEx ident;
       ident.FromBase64(base64);
+      if (!m_Storage)
+        m_Storage = GetNewStorageInstance();
+      m_Storage->AddAddress(ident);
+      m_Addresses[address] = ident.GetIdentHash();
+      LOG(info) << "AddressBook: " << address << "->"
+                << ident.GetIdentHash().ToBase32() << " added";
```

Comment:
This is *not* the same as `GetB32AddressFromIdentHash(ident.GetIdentHash())`. Look carefully.